### PR TITLE
function: add support for absent()

### PR DESCRIPTION
**Pull request by zeus2611**
Signed-off-by: Nischay [itsnischay2604@gmail.com](mailto:itsnischay2604@gmail.com)

add support for absent(), related issue https://github.com/thanos-community/promql-engine/issues/138

### CODE REVIEW

File: engine/engine_test.go [hunk 1]
```diff
@@ -337,6 +337,13 @@ func TestQueriesAgainstOldEngine(t *testing.T) {
 					http_requests_total{pod="nginx-2"} 1+2x18`,
 			query: "count_over_time(http_requests_total[30s])",
 		},
+		{
+			name: "absent",
+			load: `load 30s
+					http_requests_total{pod="nginx-1"} 0
+					http_requests_total{pod="nginx-2"} 1`,
+			query: "absent(http_requests_total[30s])",
+		},
```
[the remark this answers is not in the document]
Comment (zeus2611) [reply engine/engine_test.go:346]: i will try

**Review by GiedriusS — COMMENTED** [review]
Seems like CI failure is related.

**Comment by zeus2611** [comment]
can you suggest how i can fix this

File: execution/function/functions.go [hunk 2]
```diff
@@ -66,6 +66,21 @@ var Funcs = map[string]FunctionCall{
 	"asinh": simpleFunc(math.Asinh),
 	"acosh": simpleFunc(math.Acosh),
 	"atanh": simpleFunc(math.Atanh),
+	"absent": func(f FunctionArgs) promql.Sample{
```
Comment (yeya24) [inline execution/function/functions.go:69]: I guess we need to fix the series labels. Please refer to upstream Prometheus and see how it implements absent https://github.com/prometheus/prometheus/blob/main/promql/functions.go#L591

Comment (zeus2611) [reply execution/function/functions.go:69]: should i try to implement functions in promql-engine in similar way as promql function in prometheus

**Comment by fpetkovski** [comment]
@zeus2611 you can run `make test` locally to see where tests fail.

**Comment by zeus2611** [comment]
```
make (e=2): The system cannot find the file specified.
make: *** [Makefile:36: test] Error 2
```
This is the error that i'm getting when using `make test`

**Pull request closed without merging**